### PR TITLE
[NRH-151] Magic link auth with poc

### PR DESCRIPTION
**Pull request by michael-caktus**
Subtask of NRH-91. This implements everything but:
* Rate throttling
* Token invalidation after attempt to redeem
* Actual templated email sent
* Tests

### CODE REVIEW

File: spa/cypress/integration/contributor.spec.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      });
+    });
+
+    it('should display server generated error on email if non-200 status', () => {
+      const emailError = ['email error'];
+      cy.intercept(
+        { method: 'POST', url: getEndpoint(GET_MAGIC_LINK) },
+        { body: { email: emailError }, statusCode: 400 }
+      ).as('getMagicLink');
+      cy.getByTestId('magic-link-email-button').click();
```
Comment (benjaminEwhite) [inline spa/cypress/integration/contributor.spec.js:27]: Is there a missing assertion here? Doesn't seem like this proves that server generated error is displayed.

File: spa/src/components/contributor/ContributorEntry.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const [showConfirmation, setShowConfirmation] = useState(false);
+
+  const handleSendMagicLink = async (e) => {
+    e.preventDefault();
+    setLoading(true);
+    try {
+      const response = await axios.post(GET_MAGIC_LINK, { email });
+      if (response.status === 200) setShowConfirmation(true);
+    } catch (e) {
+      if (e?.response?.data) {
```
Comment (benjaminEwhite) [inline spa/src/components/contributor/ContributorEntry.js:28]: Is there a scenario where error doesn't have response.data? If so, doesn't look like the user would know there's a problem.

**Review by benjaminEwhite — COMMENTED** [review]
Overall looking good, but a few comments.

File: apps/contributions/models.py [hunk 3]
```diff
@@ -14,6 +17,16 @@ def contributions_count(self):
     def most_recent_contribution(self):
         return self.contribution_set.filter(status="paid").latest()
 
+    @property
+    def is_authenticated(self):
+        """
+        Copy django.contrib.auth.models import AbstractBaseUser for request.user.is_authenticated
+
+        Always return True. This is a way to tell if the user has been
```
[the remark this answers is not in the document]
Comment (michael-caktus) [reply apps/contributions/models.py:25]: Built in, called internally. I agree it sounds scary! It think it's used to differentiate between users that have an auth mechanism (based off of AbstractBaseUser) and those that don't.

**Pull request merged**